externals/kyuubi-jdbc-engine/src/main/scala/org/apache/kyuubi/engine/jdbc/operation/ExecuteStatement.scala

```diff
@@ -67,7 +67,9 @@ class ExecuteStatement(
         iter =
           if (incrementalCollect) {
             info("Execute in incremental collect mode")
-            new IterableFetchIterator(resultSetWrapper.toIterable)
+            new IterableFetchIterator[Row](new Iterable[Row] {
```

Comment:
This will cause the resultSet to be fetched lazily, so we can not close the resultSet in the finally block.

https://github.com/apache/kyuubi/blob/938384c2259b361cdf1b3bba81f615064289817c/externals/kyuubi-jdbc-engine/src/main/scala/org/apache/kyuubi/engine/jdbc/operation/ExecuteStatement.scala#L91-L94

Reply:
Thanks for the hint. It might be the cause for the failure of closing statement too early. (https://github.com/apache/kyuubi/actions/runs/5857069896/job/15886592243#step:7:292)

Let me reopen this PR to have another try.